crates/kind-query/src/dependency.rs

```diff
@@ -1,25 +1,21 @@
+use std::cell::RefCell;
+use std::hash::Hash;
+use std::rc::Rc;
+
 use fxhash::FxHashMap;
+use intmap::IntMap;
 use petgraph::{stable_graph::NodeIndex, Directed, Graph};
-use std::hash::Hash;
+
+use crate::metadata::Storage;
 
 #[derive(Default)]
-pub struct DependencyTree<Name, Data> {
-    loaded_modules: Vec<Data>,
-    loaded_names: FxHashMap<Name, (NodeIndex, usize)>,
-    graph: Graph<usize, (), Directed>,
+pub struct DependencyTree<K> {
+    pub storage: IntMap<Rc<RefCell<Option<Storage>>>>,
```

Comment:
I'm not sure if Rc<RefCell<Option<T>>> is the best option, I would just destroy the last Resource (by marking it as deleted and adding the index to another queue) and then push it to the vec.